crates/red_knot/src/types/infer.rs

```diff
@@ -145,6 +145,15 @@ fn infer_expr_type(db: &dyn SemanticDb, file_id: FileId, expr: &ast::Expr) -> Qu
     // TODO cache the resolution of the type on the node
     let symbols = symbol_table(db, file_id)?;
     match expr {
+        ast::Expr::NumberLiteral(ast::ExprNumberLiteral { value: v, .. }) => {
+            if let ast::Number::Int(n) = v {
+                // TODO support big int literals, or at least default to `builtins.int`
+                Ok(n.as_i64().map(Type::IntLiteral).unwrap_or(Type::Unknown))
```

Comment:
Could we change `Type::IntLiteral` to store a `ruff_python_ast::Int` instead of an `i64` to get support for big int literals with the optimization to avoid allocating for values that fit into `i64`?

Reply:
This caused issues for pyright as well, so it's definitely something we'll need to support eventually (but I don't think it *has* to be done in this PR):
- https://github.com/microsoft/pyright/issues/2860
- https://github.com/microsoft/pyright/issues/2879

Relevant pyright commit: https://github.com/microsoft/pyright/commit/65f444daecfa3884e082c6cd528c6bd5ee2102df

Reply:
Sure, that's a good idea, I can make this change.

Reply:
After further discussion, this would make `Type` non `Copy` which I don't want to do -- I think we will need to intern bignum literals (as well as e.g. string literals). So I'm going to leave this as-is for now.